athena-oracle/src/main/java/com/amazonaws/athena/connectors/oracle/OracleJdbcConnectionFactory.java

```diff
@@ -25,57 +25,78 @@
+    private static final String SSL_CONNECTION_STRING_REGEX = "jdbc:oracle:thin:\\$\\{([a-zA-Z0-9:_/+=.@-]+)\\}@" +
+            "\\((?i)description=\\(address=\\(protocol=tcps\\)\\(host=[a-zA-Z0-9-.]+\\)" +
+            "\\(port=([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])\\)\\)" +
+            "\\(connect_data=\\(sid=[a-zA-Z_]+\\)\\)\\(security=\\(ssl_server_cert_dn=\"[=a-zA-Z,0-9-.,]+\"\\)\\)\\)";
+    private static final Pattern SSL_CONNECTION_STRING_PATTERN = Pattern.compile(SSL_CONNECTION_STRING_REGEX);
 
     /**
      * @param databaseConnectionConfig database connection configuration {@link DatabaseConnectionConfig}
-     * @param properties               JDBC connection properties.
      * @param databaseConnectionInfo
      */
-    public OracleJdbcConnectionFactory(DatabaseConnectionConfig databaseConnectionConfig, Map<String, String> properties, DatabaseConnectionInfo databaseConnectionInfo)
+    public OracleJdbcConnectionFactory(DatabaseConnectionConfig databaseConnectionConfig, DatabaseConnectionInfo databaseConnectionInfo)
     {
-        super(databaseConnectionConfig, properties, databaseConnectionInfo);
+        super(databaseConnectionConfig, null, databaseConnectionInfo);
         this.databaseConnectionInfo = Validate.notNull(databaseConnectionInfo, "databaseConnectionInfo must not be null");
         this.databaseConnectionConfig = Validate.notNull(databaseConnectionConfig, "databaseEngine must not be null");
-        this.jdbcProperties = new Properties();
-        if (properties != null) {
-            this.jdbcProperties.putAll(properties);
-        }
     }
 
     @Override
     public Connection getConnection(final JdbcCredentialProvider jdbcCredentialProvider)
     {
         try {
             final String derivedJdbcString;
+            final Map<String, String> envVariables = System.getenv();
+            Properties properties = new Properties();
+
             if (null != jdbcCredentialProvider) {
+                Matcher connStringMatcher = SSL_CONNECTION_STRING_PATTERN.matcher(databaseConnectionConfig.getJdbcConnectionString());
+                if (OracleConstants.YES.equalsIgnoreCase(envVariables.get("ssl"))) {
+                    if (!connStringMatcher.matches()) {
+                        throw new RuntimeException("Invalid connection string to establish connection over SSL, Please check.");
+                    }
+                    properties.put("javax.net.ssl.trustStoreType", "JKS");
+                    properties.put("javax.net.ssl.trustStorePassword", "changeit");
+                    properties.put("oracle.net.ssl_server_dn_match", "true");
+
+                    LOGGER.info("Establishing connection over SSL..");
+                }
+                else {
+                    if (connStringMatcher.matches()) {
+                        throw new RuntimeException("Invalid connection string, Please check.");
```

Comment:
Actually, this makes me wonder if we need the ssl environment variable at all?

I think that since we're already pattern matching against the connection string, we can infer ssl, right?

The main reason I'd want to get rid of the extra environment variable is because the way we support SSL in the future may change if we extend beyond RDS and it may be a pain for backwards compatibility later on.

Reply:
Yes Henry, got it.

Done with the required changes, please check once.